http-client/src/test/groovy/io/micronaut/http/client/HttpGetSpec.groovy

```diff
@@ -860,17 +860,19 @@ class HttpGetSpec extends Specification {
         @Get("/queryParam")
         String queryParam(@QueryValue String foo)
 
-        @Get("/queryParamExploded{?bar*}")
-        String queryParamExploded(@QueryValue("bar") List<String> foo)
+        @Get("/queryParamExploded{?bar}")
+        String queryParamExploded(@QueryValue(value="bar", format=QueryValue.Format.MULTI) List<String> foo)
 
-        @Get("/queryParamExploded{?bar*}")
-        String queryParamExploded2(@QueryValue List<String> bar)
+        @Get("/queryParamExploded{?bar}")
+        String queryParamExploded2(@QueryValue(format=QueryValue.Format.MULTI) List<String> bar)
 
-        @Get("/multipleExplodedQueryParams{?bar*,tag}")
-        String multipleExplodedQueryParams(@QueryValue("bar") List<String> foo, @QueryValue("tag") String label)
+        @Get("/multipleExplodedQueryParams{?bar,tag}")
+        String multipleExplodedQueryParams(@QueryValue(value="bar", format=QueryValue.Format.MULTI) List<String> foo,
+                                           @QueryValue(value="tag", format=QueryValue.Format.MULTI) String label)
 
-        @Get("/multipleExplodedQueryParams{?bar*,tag}")
-        String multipleExplodedQueryParams2(@QueryValue List<String> bar, @QueryValue String tag)
+        @Get("/multipleExplodedQueryParams{?bar,tag}")
+        String multipleExplodedQueryParams2(@QueryValue(format=QueryValue.Format.MULTI) List<String> bar,
+                                            @QueryValue(format=QueryValue.Format.MULTI) String tag)
```

Comment:
Don't change existing tests. Add new ones instead

Reply:
Unless of course the old behavior should not be supported, but I don't think that is the case here

Reply:
I changed these tests because in the `{?bar*}` the `*` means the same as to specify `@QueryValue(format=QueryValue.Format.MULTI)`. Without `*` it would be `COMMA_DELIMITED` format.

For the way I implemented it, one can either ommit `@QueryValue` completely - and the parameter will be found by name and then added to query with the `UriTemplate.expand` which will parse the path template and format everything according to it.  Or the format needs to be specified inside the annotation `@QueryValue(format=...)` and then path is not read.

Now that I think of it, maybe it would be better to store the formats for query variables somewhere in the `uriContext` object, and then change the `UriTemplate.expand` to support it, as it would add the format feature everywhere instead of just for Declerative Client. But that sounds hard and I am not sure if needed.

Reply:
The existing tests need to stay as is to assert the previous behavior is maintained